packages/databases-collections-list/src/items-table.tsx

```diff
@@ -0,0 +1,604 @@
+
+              return (
+                <Fragment key={row.id}>
+                  {!isExpandedContent && (
+                    <Row
+                      className={rowStyles}
+                      data-testid={`${dataTestId}-row-${
+                        (row.original as { name?: string }).name ?? row.id
+                      }`}
+                      row={row}
+                      virtualRow={virtualRow}
+                      data-index={index}
+                      onClick={() => onItemClick(row.original._id)}
+                    >
+                      {row
+                        .getVisibleCells()
+                        .map((cell: LeafyGreenTableCell<T>) => {
+                          const isActionsCell = cell.column.id === 'actions';
+
+                          return (
+                            // cell is required
+                            <Cell
+                              key={cell.id}
+                              id={cell.id}
+                              cell={cell}
+                              className={cx({
+                                [actionsCellClassName]: isActionsCell,
+                                [actionsCellStyles]: isActionsCell,
+                              })}
+                            >
+                              {flexRender(
+                                cell.column.columnDef.cell,
+                                cell.getContext()
+                              )}
+                            </Cell>
+                          );
+                        })}
+                    </Row>
+                  )}
+                  {isExpandedContent && <ExpandedContent row={row} />}
+                </Fragment>
+              );
+            })}
+          </TableBody>
+        </Table>
+      </WorkspaceContainer>
+    </div>
+  );
+};
+
+function mapVirtualRowItems<T extends Item>(
+  table: LeafyGreenVirtualTable<T>
+): RowItem<T>[] {
+  const virtualItems = table.virtual.getVirtualItems();
+  return virtualItems.map((virtualItem) => {
+    return {
+      row: virtualItem.row,
+      virtualRow: virtualItem,
+    };
+  });
+}
+
+export const VirtualItemsTable = <T extends Item>({
+  'data-testid': dataTestId,
+  namespace,
+  itemType,
+  columns,
+  items,
+  onItemClick,
+  onDeleteItemClick,
+  onCreateItemClick,
+  onRefreshClick,
+  renderLoadSampleDataBanner,
+}: ItemsTableProps<T>): React.ReactElement => {
+  const tableContainerRef = React.useRef<HTMLDivElement>(null);
+
+  const columnsWithActions = useMemo(() => {
+    return calculateColumnsWithActions(columns, onDeleteItemClick);
+  }, [columns, onDeleteItemClick]);
+
+  const table = useLeafyGreenVirtualTable<T>({
+    containerRef: tableContainerRef,
+    data: items,
+    columns: columnsWithActions,
+    virtualizerOptions: {
+      estimateSize: () => 40,
+      overscan: 10,
+    },
+  });
+
+  const rowItems = mapVirtualRowItems(table);
+
+  return ItemsTableInner<T>({
```

Comment:
Nit: if you're just calling this as a function instead of using jsx, the name should not be capitalized to indicate that it's not really a component